R/package.r

```diff
@@ -32,66 +12,10 @@ as.package <- function(x = NULL) {
 #' \dontrun{
 #' package_file("figures", "figure_1")
 #' }
-package_file <- function(..., path = ".") {
-  if (!is.character(path) || length(path) != 1) {
-    stop("`path` must be a string.", call. = FALSE)
-  }
-  path <- strip_slashes(normalizePath(path, mustWork = FALSE))
-
-  if (!file.exists(path)) {
-    stop("Can't find '", path, "'.", call. = FALSE)
-  }
-  if (!file.info(path)$isdir) {
-    stop("'", path, "' is not a directory.", call. = FALSE)
-  }
-
-  # Walk up to root directory
-  while (!has_description(path)) {
-    path <- dirname(path)
-
-    if (is_root(path)) {
-      stop("Could not find package root.", call. = FALSE)
-    }
-  }
-
-  file.path(path, ...)
-}
-
-has_description <- function(path) {
-  file.exists(file.path(path, 'DESCRIPTION'))
-}
-
-is_root <- function(path) {
-  identical(path, dirname(path))
-}
-
-strip_slashes <- function(x) {
-  x <- sub("/*$", "", x)
-  x
+package_file <- function(path = ".", ...) {
+  file.path(pkg_path(path), ...)
```

Comment:
I thought rprojroot would handle this, but maybe not? I think `strip_slashes()` was added so that "pkgName/" worked on windows.